packager/media/formats/mp4/box_definitions.cc

```diff
@@ -1709,6 +1709,57 @@ size_t ElementaryStreamDescriptor::ComputeSizeInternal() {
   return HeaderSize() + es_descriptor.ComputeSize();
 }
 
+Mha1Specific::Mha1Specific() = default;
+Mha1Specific::~Mha1Specific() = default;
+
+FourCC Mha1Specific::BoxType() const {
+  return FOURCC_mhaC;
+}
+
+bool Mha1Specific::ReadWriteInternal(BoxBuffer* buffer) {
+
+  if (buffer->Reading()) {
+    RCHECK(ReadWriteHeaderInternal(buffer) &&
+           buffer->ReadWriteVector(
+               &extra_data, buffer->Reading() ? buffer->BytesLeft() : extra_data.size()));
+  } else {
+    uint32_t size = extra_data.size() + HeaderSize();
+    uint32_t fcc = FOURCC_mhaC;
+    buffer->ReadWriteUInt32(&size); //size of mhac + phac
+    buffer->ReadWriteUInt32(&fcc); //mhac header itself
+    RCHECK(buffer->ReadWriteVector(&extra_data, extra_data.size()));
+  }
+
+  return true;
+}
+
+size_t Mha1Specific::ComputeSizeInternal() {
+  if (extra_data.empty())
+    return 0;
+  return HeaderSize() + extra_data.size();
+}
+
+Mhm1Specific::Mhm1Specific() = default;
+Mhm1Specific::~Mhm1Specific() = default;
+
+FourCC Mhm1Specific::BoxType() const {
+  return FOURCC_mhm1;
+}
+
+bool Mhm1Specific::ReadWriteInternal(BoxBuffer* buffer) {
+  RCHECK(ReadWriteHeaderInternal(buffer) &&
```

Comment:
Here is the definition of `MHADecoderConfigurationRecord`:

```
aligned(8) class MHADecoderConfigurationRecord {
unsigned int(8) configurationVersion = 1;
unsigned int(8) mpegh3daProfileLevelIndication;
unsigned int(8) referenceChannelLayout;
unsigned int(16) mpegh3daConfigLength;
bit(8*mpegh3daConfigLength) mpegh3daConfig;
}
```

mpegh3daProfileLevelIndication will be needed for the codec string, so it should be extracted.

It can be done by adding one extra instruction after line 1752:

```
  if (!buffer->Reading()) {
    RCHECK(data.size() > 1);
    mpeg_h_3da_profile_level_indication = data[1];
  }
```

Reply:
I've set `mpeg_h_3da_profile_level_indication`, but what should I do with it? Is it read or written somewhere?

Reply:
See the comment above in packager/media/base/audio_stream_info.cc, it should be used there to set the codec string.

Reply:
Gotcha. In that case, how does mpeg_h_3da_profile_level_indication get passed to object_audio_type?

Reply:
It can be set in mp4_media_parser.cc, e.g.

```
case FOURCC_mha1:
case FOURCC_mhm1:
  codec_config = entry.mhac.extra_data;
  audio_object_type = entry.mhac. mpeg_h_3da_profile_level_indication;
  break;
```

Reply:
Gotcha. Addressed in this pull request: https://github.com/google/shaka-packager/pull/952/files

It's strange though. I'm able to get the leveling info from `data[1]` (i.e. `0c`, `0d`, `0e`), but it is not getting written to the codec string via `audio_object_type`.

Reply:
Okay, was able to get it resolved. Should be included in #952